nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,502 @@
+              :obj:`list` of :obj:`~nilearn.surface.SurfaceImage` or \
+              :obj:`tuple` of :obj:`~nilearn.surface.SurfaceImage`
+            Mesh and data for both hemispheres. The data for each hemisphere \
+            is of shape (n_vertices/2, n_timepoints).
+
+        y : None
+            This parameter is unused.
+            It is solely included for scikit-learn compatibility.
+
+        confounds : :class:`numpy.ndarray`, :obj:`str`,\
+                    :class:`pathlib.Path`, \
+                    :class:`pandas.DataFrame` \
+                    or :obj:`list` of confounds timeseries, default=None
+            Confounds to pass to :func:`nilearn.signal.clean`.
+
+        sample_mask : None, Any type compatible with numpy-array indexing, \
+                  or :obj:`list` of \
+                  shape: (number of scans - number of volumes removed) \
+                  for explicit index, or (number of scans) for binary mask, \
+                  default=None
+            sample_mask to pass to :func:`nilearn.signal.clean`.
+
+
+        Returns
+        -------
+        region_signals: :obj:`numpy.ndarray`
+            Signal for each region as provided in the maps (via `maps_img`).
+            shape: (n_timepoints, n_regions)
+        """
+        del y
+        return self.fit().transform(img, confounds, sample_mask)
+
+    def inverse_transform(self, region_signals):
+        """Compute :term:`vertex` signals from region signals.
+
+        Parameters
+        ----------
+        region_signals: :obj:`numpy.ndarray`
+            Signal for each region as provided in the maps (via `maps_img`).
+            shape: (n_timepoints, n_regions)
+
+        Returns
+        -------
+        vertex_signals: :obj:`~nilearn.surface.SurfaceImage`
+            Signal for each vertex projected on the mesh of the `maps_img`.
+            The data for each hemisphere is of shape
+            (n_vertices/2, n_timepoints).
+        """
+        self._check_fitted()
+
+        if region_signals.shape[1] != self.n_elements_:
+            raise ValueError(
+                f"Expected {self.n_elements_} regions, "
+                f"but got {region_signals.shape[1]}."
+            )
+
+        # project region signals back to vertices
+        vertex_signals = np.dot(region_signals, self.maps_img_.T)
+
+        # we need the data to be of shape (n_vertices, n_timepoints)
+        # because the SurfaceImage object expects it
+        vertex_signals = vertex_signals.T
+
+        # split the signal into hemispheres
+        vertex_signals = {
+            "left": vertex_signals[
+                : self.maps_img.data.parts["left"].shape[0], :
+            ],
+            "right": vertex_signals[
+                : self.maps_img.data.parts["right"].shape[0], :
+            ],
+        }
+
+        return SurfaceImage(mesh=self.maps_img.mesh, data=vertex_signals)
+
+    def generate_report(self, displayed_maps=10):
```

Comment:
@man-shu 

should we keep the report part for another PR?

Would make it easier to review (shorter PR) and we seem to have some report issues to fix anyway?

Reply:
Oops you are right! We are at >800 lines as of now. Will try to figure out how I can move some of the commits to another PR

Reply:
A lot of them are the HTML report btw